README.md

```diff
@@ -7,7 +7,112 @@
 [![crates.io](https://img.shields.io/crates/v/stm32f1xx-hal.svg)](https://crates.io/crates/stm32f1xx-hal)
 [![Released API docs](https://docs.rs/stm32f1xx-hal/badge.svg)](https://docs.rs/stm32f1xx-hal)
 
-## Usage
+## Quick start guide
+
+Embedded rust development requires a bit more setup than ordinary development.
+For this guide, we'll assume you're using a stm32 blue pill board (shown
+below), but if you have another f1 microcontroller, you should be able to adapt
+it.
+
+![blue pill pinout](BluePillPinout.jpg "opt title")
+
+You will also need an [stlink](https://www.adafruit.com/product/2548) for
```

Comment:
This looks like a rebranded Chinesium ST-Link clone with pirated firmware (not sure how they can offer that on in the adafruit line). Can we recommend a genuine Product instead (like the ST-Link V3 Mini) which is even cheaper, though quite a bit trickier to obtain, insteads?

Reply:
Alternatively we could recommend an open product or maybe hint to an official Nucleo board which happens to include an ST-Link so works out-of-the-box.

Reply:
I think we should use something compatible with blue pills since they seem pretty common in the rust community.

Recommending a better alternative to the weird stlink sounds like a good idea, got a link to one?

Reply:
Any debug probe should be compatible with the blue pill. The question is only whether we want to stay with a native ST solution or recommend something else. The ST-Link v3 mini can be found here: https://www.st.com/en/development-tools/stlink-v3mini.html